### ISSUE TITLE
Doing a bookmarks restore after a Firefox Sync fail breaks Livemarks

### ISSUE THREAD

**Issue by JatB** [nt1m/livemarks#145]
**Add-on version**
This was in Livemarks 1.6 and Firefox 64.0.2.

**Describe the bug**
This started with a Firefox Sync issue that I experience rather often. If I login to a computer that I haven't used in a couple months with Firefox Sync turned on, most of my bookmarks will become a complete mess with duplicates all over the place. If I went back to my main computer, Sync had also messed up its bookmarks. In the past, I fixed this on my main computer by doing a restore of the bookmarks from the previous restore point.

I tried that today, and all of my Livemarks were still there but I noticed they were not refreshing. When I opened the Livemarks preference, almost all of them were not there. So it appears doing a restore of the bookmarks breaks the reference to them in Livemarks.

What's strange is that I tried restoring the entire Firefox profile to the previous day using Time Machine, rebooting, and the Livemarks would still not work.

**To Reproduce**
1. You need Firefox Sync to mess up your bookmarks by trying to sync with a computer that has been off for a month or two.
2. You might also be able to try it by just restoring from a previous bookmarks restore point. Either way, you have to do a restore of a previous bookmarks save point.
3. Livemarks will no longer be connected or refresh. The "Open 'site'" and separator at the top will of the feed will be out of place.

**Expected behavior**
Livemarks to continue to work after doing a bookmarks restore.

**Extra context**
I know you recently did some work trying to get Firefox Sync support to work. I'm not sure if this issue has anything to do with that. Also, to finally get Livemarks to work again, I had to delete all of them and then add them back again.

**Comment by lukecz81** [comment]
Same problem here

**Comment by JatB** [comment]
The only real workaround I found was to turn off bookmarks syncing to my Firefox account. It's a drag, but it kept happening and would take a lot of effort to fix.

Part of the problem is that Livemarks doesn't let you easily remove multiple broken links.

The main fault is with Mozilla, of course. After all these years they still haven't fixed bookmarks syncing (just look on their Bugzilla and you will see tons of people complaining about the issue on bugs that have been there for years untouched). They are also the ones that removed Live Bookmarks and really started this whole problem.

**Comment by lukecz81** [comment]
Thank you for answer.
I can't remove ANY of broken link - there is only option to assign folder to it.
Even if I remove Livemarks extension, delete my RSS folder (folder set in Livemark's options) and reinstall Livermarks - there are still broken links - do you know in which file are these links stored to delete them manually?

**Comment by JatB** [comment]
Nope, there may be a way, but as you can see, no one has responded to this issues besides you.

What I've had to do in the past is to assign each broken livemark to a folder in Livemarks  and then delete it in the Bookmarks Library. That's what I meant by "a lot of effort to fix".

If they can't fix the issue, it would be nice if they added the option to clear broken livemarks or be able to select multiple livemarks and delete them.

I've attached a couple screenshots so people can see the issue we are talking about.

![Broken livemarks](https://user-images.githubusercontent.com/10157440/65665788-30c77080-dff1-11e9-82f6-eeb801fe0d43.png)

![Select folder](https://user-images.githubusercontent.com/10157440/65665846-49378b00-dff1-11e9-9565-edf731aac7aa.png)

One thing I have noticed that is weird, is that livemarks moved to a subfolder will not break. Or I should say, it will create broken duplicates, but there will also remain working versions. That's what you see with four of the livemarks in my screenshot.

**Comment by lukecz81** [comment]
Yes, I'm doing the same now :)
I've created a lot of new folders, and assign broken links one by one, and then delete.
When I clean it I will add my feeds, one by one.
But I'm afraid, when I turn on my second computer, the newly created links will be broken again :(

**Comment by JatB** [comment]
I think what I did was make sure everything was setup how I wanted on my main computer. Once that was the case, exported the bookmarks in the Bookmarks Library and exported the livemarks in Livemarks-More Options-Export feeds.

I then made sure my other computers were completely offline, opened Firefox, and turned off Bookmarks syncing under Preferences-Sync-Sync Settings-Bookmarks.

I then deleted all bookmarks on the second computer so there wouldn't be duplicates and imported the bookmarks from my new file. I'm pretty sure I deleted all of the livemark bookmarks so there wouldn't be any conflict and then imported the OPML file in Livemarks. I also think I had to put them in the correct order because that wasn't kept for some reason in Livemarks.

Like I said, a lot of effort for something that should work automatically in the background and users should not expect to have to deal with.
